### PR TITLE
Mx 14670 collect outgoing sovereign logs

### PR DESCRIPTION
**Pull request by mariusmihaic**
## Reasoning behind the pull request
- Added `OutgoingOperationsFormatter` which collects relevant outgoing events for bridge from the logs and creates an outgoing data that needs to be signed by validators to bridge tokens
  
## Proposed changes
- 
- 
- 

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: process/block/sovereignChainBlock.go [hunk 1]
```diff
@@ -37,23 +39,31 @@ type sovereignChainBlockProcessor struct {
 	extendedShardHeaderTracker   extendedShardHeaderTrackHandler
 	extendedShardHeaderRequester extendedShardHeaderRequestHandler
 	chRcvAllExtendedShardHdrs    chan bool
+	outgoingOperationsFormatter  sovereign.OutgoingOperationsFormatter
+}
+
+type ArgsSovereignChainBlockProcessor struct {
```
Comment (mariusmihaic) [inline process/block/sovereignChainBlock.go:45]: Comment for this is already added in  https://github.com/multiversx/mx-chain-go/pull/5673

File: process/block/sovereignChainBlock.go [hunk 2]
```diff
@@ -840,11 +850,19 @@ func (scbp *sovereignChainBlockProcessor) processSovereignBlockTransactions(
 		return nil, err
 	}
 
+	// TODO: Marius C.: Implement signing this message
+	_ = scbp.collectOutGoingOperations()
```
Comment (raduchis) [inline process/block/sovereignChainBlock.go:854]: no need for this message to be added in the block?

File: process/block/sovereignChainBlock_test.go [hunk 3]
```diff
@@ -61,7 +66,11 @@ func TestSovereignBlockProcessor_NewSovereignChainBlockProcessorShouldWork(t *te
 
 		arguments := CreateMockArguments(createComponentHolderMocks())
 		sp, _ := blproc.NewShardProcessor(arguments)
-		scbp, err := blproc.NewSovereignChainBlockProcessor(sp, nil)
+		scbp, err := blproc.NewSovereignChainBlockProcessor(blproc.ArgsSovereignChainBlockProcessor{
+			ShardProcessor:               sp,
+			ValidatorStatisticsProcessor: nil,
+			OutgoingOperationsFormatter:  &sovereign.OutgoingOperationsFormatterStub{},
```
Comment (raduchis) [inline process/block/sovereignChainBlock_test.go:72]: please add a test for OutgoingOperationsFormatter = nil

**Review by raduchis — CHANGES_REQUESTED** [review]

File: process/block/sovereign/errors.go [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package sovereign
+
+import "errors"
+
+var errNoSubscribedAddresses = errors.New("no subscribed addresses provided")
```
[the remark this answers is not in the document]
Comment (mariusmihaic) [reply process/block/sovereign/errors.go:5]: 1. I don't use a `_test` package for testing so there is no problem for this. Even if I used, I could've created an `export_test.go` for them
2. I've created a separate `sovereign` package in the block processor package where I want to "contain" all necessary sovereign related without exporting them in other packages such as `errors/errors.go`

File: process/block/sovereign/outgoingOperationsFactory.go [hunk 5]
```diff
@@ -0,0 +1,62 @@
+package sovereign
+
+import (
+	"fmt"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-go/config"
+)
+
+// CrateOutgoingOperationsFormatter creates an outgoing operations formatter
```
[the remark this answers is not in the document]
Comment (mariusmihaic) [reply process/block/sovereign/outgoingOperationsFactory.go:10]: Right, good catch, fixed it

File: process/block/sovereign/outgoingOperations.go [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+	Identifier []byte
+	Addresses  map[string]string
+}
+
+type outgoingOperations struct {
+	subscribedEvents []SubscribedEvent
+	roundHandler     RoundHandler
+}
+
+// TODO: We should create a common base functionality from this component. Similar behavior is also found in
```
[the remark this answers is not in the document]
Comment (mariusmihaic) [reply process/block/sovereign/outgoingOperations.go:28]: :pray:

File: process/block/sovereignChainBlock.go (hunk 2 above)
Comment (mariusmihaic) [reply process/block/sovereignChainBlock.go:854]: It is going to be added in next PR here: https://github.com/multiversx/mx-chain-go/pull/5673/files#diff-6af228a15da3fe0b40476a5a20ade8f283e2be09bc144907650a055a8824c69fR863

File: process/block/sovereignChainBlock_test.go (hunk 3 above)
Comment (mariusmihaic) [reply process/block/sovereignChainBlock_test.go:72]: Already added test in next PR here: https://github.com/multiversx/mx-chain-go/pull/5673/files#diff-c6abeee02153644592d72054a78cc21503eb6607028caaf7df1a79c364ba59b8R110

**Review by raduchis — APPROVED** [review]
